src/blocks/weather/nws.rs

```diff
@@ -0,0 +1,380 @@
+            (self.temperature.value - 32.0) * 5.0 / 9.0
+        };
+        let humidity = self.relative_humidity.value;
+        let wind_speed = self.wind_kmh();
+        australian_apparent_temp(temp, humidity, wind_speed)
+    }
+
+    fn to_moment(&self) -> WeatherMoment {
+        let icon = short_forecast_to_icon(&self.short_forecast, !self.is_daytime);
+        let weather = Self::icon_to_word(icon);
+        WeatherMoment {
+            icon,
+            weather,
+            weather_verbose: self.short_forecast.clone(),
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: Some(self.wind_direction()),
+        }
+    }
+
+    fn to_aggregate(&self) -> ForecastAggregate {
+        ForecastAggregate {
+            temp: self.temperature.value,
+            apparent: self.apparent_temp(),
+            humidity: self.relative_humidity.value,
+            wind: self.wind_speed.value,
+            wind_kmh: self.wind_kmh(),
+            wind_direction: Some(self.wind_direction()),
+        }
+    }
+}
+
+fn combine_forecasts(data: &[ForecastAggregate], fin: WeatherMoment) -> Forecast {
+    let mut temp = 0.0;
+    let mut apparent = 0.0;
+    let mut humidity = 0.0;
+    let mut wind_north = 0.0;
+    let mut wind_east = 0.0;
+    let mut wind_kmh_north = 0.0;
+    let mut wind_kmh_east = 0.0;
+    let mut max = ForecastAggregate {
+        temp: -1000.0,
+        apparent: -1000.0,
+        humidity: 0.0,
+        wind: 0.0,
+        wind_kmh: 0.0,
+        wind_direction: Some(0.0),
+    };
+    let mut min = ForecastAggregate {
+        temp: 1000.0,
+        apparent: 1000.0,
+        humidity: 100.0,
+        wind: 1000.0,
+        wind_kmh: 1000.0,
+        wind_direction: Some(0.0),
+    };
+    for val in data {
+        // Summations for averaging
+        temp += val.temp;
+        apparent += val.apparent;
+        humidity += val.humidity;
+        let (sin, cos) = val
+            .wind_direction
+            .unwrap_or_default()
+            .to_radians()
+            .sin_cos();
+        wind_north += val.wind * cos;
+        wind_east += val.wind * sin;
+        wind_kmh_north += val.wind_kmh * cos;
+        wind_kmh_east += val.wind_kmh * sin;
+
+        // Max
+        max.temp = max.temp.max(val.temp);
+        max.apparent = max.apparent.max(val.apparent);
+        max.humidity = max.humidity.max(val.humidity);
+        if val.wind > max.wind {
+            max.wind_direction = val.wind_direction;
+        }
+        max.wind = max.wind.max(val.wind);
+        max.wind_kmh = max.wind_kmh.max(val.wind_kmh);
+
+        // Min
+        min.temp = min.temp.min(val.temp);
+        min.apparent = min.apparent.min(val.apparent);
+        min.humidity = min.humidity.min(val.humidity);
+        if val.wind > min.wind {
```

Comment:
Should be <

Also the wind speeds can be set inside the if statement, without using the min/max functions.

Reply:
Fixed